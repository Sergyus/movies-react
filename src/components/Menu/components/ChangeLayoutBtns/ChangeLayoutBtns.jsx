import React, { useState, useCallback } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import styles from './ChangeLayoutBtns.scss';

const ChangeLayoutBtns = ({ changeLayout }) => {
  const [active, setActive] = useState('grid');
  return (
    <div>
      <button
        aria-label="Grid"
        onClick={useCallback(() => {
          setActive('grid');
          changeLayout('grid');
        }, [changeLayout])}
        className={cn(styles.btn, { [`${styles.btn__active}`]: active === 'grid' })}
        type="button"
      />
      <button
        aria-label="Table"
        onClick={useCallback(() => {
          setActive('table');
          changeLayout('table');
        }, [changeLayout])}
        className={cn(styles.btn, { [`${styles.btn__active}`]: active === 'table' })}
        type="button"
      />
    </div>
  );
};

ChangeLayoutBtns.propTypes = {
  changeLayout: PropTypes.func,
};

ChangeLayoutBtns.defaultProps = {
  changeLayout: () => {},
};

export default ChangeLayoutBtns;
