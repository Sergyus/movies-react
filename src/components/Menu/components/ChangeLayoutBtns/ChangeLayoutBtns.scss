@import "../../../../styles/variables";

.btn {
  cursor: pointer;
  border: none;
  outline: none;
  background-color: transparent;
  transition: background-color 0.3s;

  &:first-child {
    margin-right: 10px;
    width: 20px;
    height: 20px;
    -webkit-mask: url(("../../../../assets/images/sprites.png")) no-repeat -170px -30px;
    mask: url(("../../../../assets/images/sprites.png")) no-repeat -170px -30px;
    background-color: $gray-color;
  }

  &:last-child {
    margin-right: 10px;
    width: 20px;
    height: 20px;
    -webkit-mask: url(("../../../../assets/images/sprites.png")) no-repeat -30px -170px;
    mask: url(("../../../../assets/images/sprites.png")) no-repeat -30px -170px;
    background-color: $gray-color;
  }

  &:hover {
    background-color: $blue-color;
  }
}

.btn.btn__active {
  background-color: $blue-color;
}

@media (max-width: $tablet-screen-size) {
  .btn {
    margin-bottom: 20px;
  }
}
