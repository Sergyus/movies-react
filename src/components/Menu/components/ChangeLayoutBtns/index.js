import ChangeLayoutBtnsContainer from './ChangeLayoutContainer';

export default ChangeLayoutBtnsContainer;
