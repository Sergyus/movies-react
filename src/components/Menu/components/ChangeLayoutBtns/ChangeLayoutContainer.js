import { connect } from 'react-redux';
import ChangeLayoutBtns from './ChangeLayoutBtns';
import { changeLayout } from '../../../../modules/ui';

export default connect(null, { changeLayout })(ChangeLayoutBtns);
