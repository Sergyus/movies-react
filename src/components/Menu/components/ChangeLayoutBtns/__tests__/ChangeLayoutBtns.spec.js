import React from 'react';
import { create, act } from 'react-test-renderer';
import ChangeLayoutBtns from '../ChangeLayoutBtns';

describe('ChangeLayoutBtns component', () => {
  it('ChangeLayoutBtns renders correctly', () => {
    const tree = create(<ChangeLayoutBtns />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Grid button set active', () => {
    const tree = create(
      <ChangeLayoutBtns />,
    );
    const button = tree.root.findByProps({ 'aria-label': 'Grid' });
    act(() => {
      button.props.onClick();
    });
    expect(button.props.className).toEqual('btn btn__active');
  });

  it('Table button set active', () => {
    const tree = create(
      <ChangeLayoutBtns />,
    );
    const button = tree.root.findByProps({ 'aria-label': 'Table' });
    act(() => {
      button.props.onClick();
    });
    expect(button.props.className).toEqual('btn btn__active');
  });
});
