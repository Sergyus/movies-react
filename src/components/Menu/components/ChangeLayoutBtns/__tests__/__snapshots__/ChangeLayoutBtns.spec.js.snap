// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeLayoutBtns component ChangeLayoutBtns renders correctly 1`] = `
<div>
  <button
    aria-label="Grid"
    className="btn btn__active"
    onClick={[Function]}
    type="button"
  />
  <button
    aria-label="Table"
    className="btn"
    onClick={[Function]}
    type="button"
  />
</div>
`;
